src/get-started/fwe/local-caching.md

```diff
@@ -0,0 +1,232 @@
+All caching strategies run the risk of holding onto stale data.
+Unfortunately, the action of verifying the freshness of a cache
+often takes as much time to complete as fully loading the data
+in question. This means that most apps tend to only benefit
+from caching data if they trust the data to be fresh at runtime
+without verification.
+
+To deal with this, most caching systems include a time limit
+on any individual piece of cached data. After this time limit
+is exceeded, would-be cache hits are treated as cache misses
+until fresh data is loaded.
+
+A popular joke among computer scientists is that "The two
+hardest things in computer science are cache invalidation,
+naming things, and off-by-one errors." 😄
+
+Despite the risks, almost every app in the world makes heavy
+use of data caching. The rest of this page explores multiple
+approaches to caching data in your Flutter app, but know that
+all of these approaches can be tweaked or combined for your
+situation.
+
+## Caching data in local memory
+
+The simplest and most performant caching strategy is an
+in-memory cache. The downside of this strategy is that,
+because the cache is only held in system memory, no data is
+retained beyond the session in which it is originally cached.
+(Of course, this "downside" also has the upside of automatically
+solving most stale cache problems!)
+
+Due to their simplicity, in-memory caches closely mimic
+the pseudocode seen above. That said, it is best to use proven
+design principles, like the [repository pattern][],
+to organize your code and prevent cache checks like the above
+from appearing all over your code base.
+
+Imagine a `UserRepository` class that is also tasked with
+caching users in memory to avoid duplicate network requests.
+Its implementation might look like this:
+
+```dart
+class UserRepository {
+  UserRepository(this.api);
+  
+  final Api api;
+  final Map<int, User?> _userCache = {};
+
+  Future<User?> loadUser(int id) async {
+    if (!_userCache.containsKey(id)) {
+      final response = await api.get(id);
+      if (response.statusCode == 200) {
+        _userCache[id] = User.fromJson(response.body);
+      } else {
+        _userCache[id] = null;
+      }
+    }
+    return _userCache[id];
+  }
+}
+```
+
+This `UserRepository` follows multiple proven design
+principles including:
+
+* [dependency injection][], which helps with testing
+* [loose coupling][], which protects surrounding code from
+its implementation details, and
+* [separation of concerns][], which prevents its implementation
+from juggling too many concerns.
+
+And best of all, no matter how many times within a single session
+a user visits pages in your Flutter app that load a given user,
+the `UserRepository` class only loads that data over the network *once*.
+
+However, your users might eventually tire of waiting for data
+to load every time they relaunch your app. For that, you should
+choose from one of the persistent caching strategies found below.
+
```

Comment:
Probably out of the scope, but the Proxy design pattern is used quite often for this specific use case. It could be provided as a code architecture alternative for the in-memory cache. Resources:
https://flutterdesignpatterns.com/pattern/proxy
https://kazlauskas.dev/blog/flutter-design-patterns-15-proxy/